jiratools/formatting.py

```diff
@@ -1,13 +1,18 @@
 """Formatting helpers for jiratools."""
 import getpass
 import os
+from typing import Optional, List, Union
 
+DataArray = List[List[Union[str, int]]]
 
-def _build_source():
+
+def _build_source() -> str:
     return os.environ.get("BUILD_URL", "Manual run by {}".format(getpass.getuser()))
 
 
-def format_autoupdate_jira_msg(message_body, header_body=None):
+def format_autoupdate_jira_msg(
+    message_body: str, header_body: Optional[str] = None
```

Comment:
Just an opinion here: I don't like the `Optional` thing. It isn't optional. You have to pass in something, or Python will error out. But you _can_ pass in `None`. E > I is pythonic, so I think `Union[None, str]` is more explicit and accurate.

Reply:
Sorry I missed the `= None`, so I revert my comment.

Reply:
I do kind of wish that `= None` meant it could be `None` or whatever else I put. I can see the reasoning for it though.

Reply:
In this particular case, it is _truly_ optional:
```
    message = "h2. {}".format(header_body) if header_body else ""
    message += "\n\nAutomated JIRA Update:\n\n{}\n\n{}".format(
        _build_source(), message_body
    )
    return message
```

we explicitly handle the option to either provide a `header_body` or not, and format the message appropriately either way.